encoding/protobuf/protoc-gen-yarpc-go/internal/lib/lib.go

```diff
@@ -193,6 +193,81 @@ func Build{{$service.GetName}}YARPCProcedures(server {{$service.GetName}}YARPCSe
 	)
 }
 
+// Provide{{$service.GetName}}YARPCClientParams defines the input
+// for Provide{{$service.GetName}}YARPCClient. It provides the
+// paramaters to get a {{$service.GetName}}YARPCClient in an
+// Fx application.
+type Provide{{$service.GetName}}YARPCClientParams struct {
+	fx.In
+
+	Provider yarpc.ClientConfig
+}
+
+// Provide{{$service.GetName}}YARPCClientResult defines the output
+// of Provide{{$service.GetName}}YARPCClient. It provides a
+// {{$service.GetName}}YARPCClient to an Fx application.
+type Provide{{$service.GetName}}YARPCClientResult struct {
+	fx.Out
+
+	Client {{$service.GetName}}YARPCClient
+
+	// We are using an fx.Out struct here instead of just returning a client
+	// so that we can add more values or add named versions of the client in
+	// the future without breaking any existing code.
+}
+
+// Provide{{$service.GetName}}YARPCClient provides a {{$service.GetName}}YARPCClient
+// to an Fx application using the given name for routing.
+//
+//  fx.Provide(
+//    {{$packageName}}.Provide{{$service.GetName}}YARPCClient,
+//    ...
+//  )
+func Provide{{$service.GetName}}YARPCClient(name string, options ...protobuf.ClientOption) interface{} {
```

Comment:
If something is named `ProvideFoo`, the expectation is that it will return an
`fx.Option`, not a thing that you'll provide.

So this should either be renamed to just `func New{{serviceName}}YARPCClient`
or similar, or return an `fx.Option` by calling `fx.Provide`.

The former is preferable largely because it'll be confusing otherwise to do,

    fx.New(
      protopackage.ProvideFooYARPCClient("foo"),
      fx.Provide(
        thriftpackage.BarClient("bar")
      ),
    )

(You can't `fx.Provide` an `fx.Option`.)

Reply:
We already have `New{{serviceName}}YARPCClient`, this is how you construct Clients without Fx. What alternative naming would you suggest?

Reply:
That naming issue is why the Fx module gets its own subpackage. There's no
precedent here on whether Protobuf Fx modules get their own package or not. If
they get their own package, there's a lot more room around what it's called.

If it **has** to be in this package, there are two options:

- `ProvideFoo` returns an `fx.Option`
- `SomethingFooYARPCClient` where `Something` is TBD returns `interface{}`

What `Something` is, I don't know. I'm asking for some more deliberation
around Fx module structure for Proto before we commit to this.

Again, typically constructors are named `NewFoo` as in regular Go code but
that name is already taken -- which is why the Fx module was in a subpackage
in the first place.

Reply:
Should we schedule a meeting to decide on what option to take?

This should be in the same package, this is the standard Protobuf plugin convention. I'd like to decide this however, if a meeting is the quickest path, then let's set that up.